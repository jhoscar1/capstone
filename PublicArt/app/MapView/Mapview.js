import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image
} from 'react-native';
import MapView from 'react-native-maps';
import MapCallout from './MapCallout';
import { StackNavigator } from 'react-navigation';
import PointDetails from '../PointDetails';

class Mapview extends Component {
  constructor(props){
    super(props)
    this.state = {
      selectedMarker: false,
    };
    this.markerSelected = this.markerSelected.bind(this)
  }

  markerSelected (markerId) {
    this.setState({
      selectedMarker: markerId
    });
  }

  render() {
    console.log("map props when rendering: ", this.props)
    let region = {};

    const navigation = this.props.navigation

    if (this.props.screenProps.position.coords && this.props.screenProps.allPois.length) {
      let userLocation = this.props.screenProps.position.coords
      let markerArr = (this.props.navigation.state.params) ? this.props.navigation.state.params.markers
                                                           : this.props.screenProps.allPois
        region = {
          latitude: (markerArr.length==1) ? +markerArr[0].lat : +userLocation.latitude,
          longitude: (markerArr.length==1) ? +markerArr[0].lng : +userLocation.longitude,
          latitudeDelta: 0.0122,
          longitudeDelta: 0.0021,
        }
        console.log("MARKERS: ", markerArr)
        return (
          <View style={styles.container}>
          <MapView
          style={ styles.map }
            initialRegion={region}>
              <MapView.Marker
                style={ styles.personMarker }
                coordinate={{
                  latitude: this.props.screenProps.position.coords.latitude,
                  longitude: this.props.screenProps.position.coords.longitude
                }}
                image={require('../../PersonIcon.png')}
              />
              {
                markerArr.map((marker, index) => (
                marker.lat > 0 ?
                   <MapView.Marker
                      key={marker.unique_id}
                      coordinate={{
                        latitude: +marker.lat,
                        longitude: +marker.lng}}
                      id={marker.unique_id}
                      onPress={(event) => this.markerSelected(marker.unique_id)}>
                        <MapView.Callout tooltip style={styles.customView}
                          onPress={() => navigation.navigate('Details', { name: marker.link})}>
                          {marker.unique_id === this.state.selectedMarker ?
                          <MapCallout>
                            <Text
                              style={ styles.titleText }
                              >{marker.name}</Text>
                            <Image
                              source={{uri: `https://www.nycgovparks.org${marker.thumb_path}`}}
                              style={styles.placeImage}/>
                            <Text
                              style={ styles.titleText }>Tap for Info</Text>
                          </MapCallout>
                           :
                          null}
                        </MapView.Callout>
                  </MapView.Marker>
                  : null
              ))
            }
            </MapView>
          </View>
        )
      }
      else return null
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  map: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  personMarker: {
    zIndex: 1,
    position: 'relative'
  },
  titleText: {
    fontWeight: 'bold',
    textAlign: 'center'
  },
  tapText: {
    textDecorationLine: 'underline',
    textAlign: 'center',
    color: '#0000ff'
  },
  placeImage: {
    height: 75,
    width: 75,
    alignSelf: 'center'
  },
  customView: {
    height: 100,
    width: 140,
  }
});

const MapStack = StackNavigator({
  MainMap: {
    screen: Mapview,
    title: "Map View"
  },
  MapDetails: {
    screen: PointDetails,
    path: 'poi/:name',
    title: "Details"
  }
})

// AppRegistry.registerComponent('MapStack', () => MapStack);

export default MapStack;