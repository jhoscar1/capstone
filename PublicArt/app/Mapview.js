import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Button,
  Text,
  Image
} from 'react-native';
import MapView from 'react-native-maps';
import MapCallout from './MapCallout';

class Mapview extends Component {
  constructor(props){
    super(props)
    const userLocation = this.props.navigation.state.params.userLocation.coords
    this.state = {
      region: {
        latitude: userLocation.latitude,
        longitude: userLocation.longitude,
        latitudeDelta: 0.0522,
        longitudeDelta: 0.0061,
       },
      markers: this.props.navigation.state.params.markers,
      selectedMarker: false
    };
    this.markerSelected = this.markerSelected.bind(this)
  }

markerSelected (markerId) {
  this.setState({
    selectedMarker: markerId
  });
}

  render() {
    const navigation = this.props.navigation
    return (
      <View style={styles.container}>
      <MapView
      style={ styles.map }
        initialRegion={this.state.region}>
          <MapView.Marker
            style={ styles.personMarker }
            coordinate={{
              latitude: this.state.region.latitude,
              longitude: this.state.region.longitude
            }}
            image={require('../PersonIcon.png')}
          />
          {this.state.markers.map((marker, index) => (
            marker.lat > 0 ?
               <MapView.Marker
                  key={marker.unique_id}
                  coordinate={{
                    latitude: +marker.lat,
                    longitude: +marker.lng}}
                  id={marker.unique_id}
                  onPress={(event) => this.markerSelected(marker.unique_id)}>
                    <MapView.Callout tooltip style={styles.customView}
                      onPress={() => navigation.navigate('Details', { name: marker.link})}>
                      {marker.unique_id === this.state.selectedMarker ?
                      <MapCallout>
                        <Text
                          style={ styles.titleText }
                          >{marker.name}</Text>
                        <Image
                          source={{uri: `https://www.nycgovparks.org${marker.thumb_path}`}}
                          style={styles.placeImage}/>
                        <Text
                          style={ styles.titleText }>Tap for Info</Text>
                      </MapCallout>
                       :
                      null}
                    </MapView.Callout>
              </MapView.Marker>
              : null
          ))}
        </MapView>
      </View>
    );
  }
  }

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  map: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  personMarker: {
    zIndex: 1,
    position: 'relative'
  },
  titleText: {
    fontWeight: 'bold',
    textAlign: 'center'
  },
  tapText: {
    textDecorationLine: 'underline',
    textAlign: 'center',
    color: '#0000ff'
  },
  placeImage: {
    height: 75,
    width: 75,
    alignSelf: 'center'
  }
});

export default Mapview;
