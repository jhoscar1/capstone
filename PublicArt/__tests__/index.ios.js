import 'react-native';
import React from 'react';
import Mapview from '../app/Mapview.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { navigation } from './TestData.js'

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});

it('renders correctly', () => {
  const component = renderer.create(
    <Mapview navigation={navigation} />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

it('renders callout correctly', () => {
  const component = renderer.create(
    <Mapview navigation={navigation} />
  )
  component.toJSON().children[0].children[1].props.onPress()
  const instance = component.getInstance()
  const stateMarkerID = instance.state.selectedMarker
  const calloutID = component.toJSON().children[0].children[1].props.id

  expect(stateMarkerID).toEqual(calloutID);
});


// number of markers is 2

// press marker to launch callout
