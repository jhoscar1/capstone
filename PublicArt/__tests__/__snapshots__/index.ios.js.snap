// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 1`] = `1`;

exports[`renders callout correctly 1`] = `undefined`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <MapView
    initialRegion={
      Object {
        "latitude": 40.729713,
        "latitudeDelta": 0.0122,
        "longitude": -73.9915285,
        "longitudeDelta": 0.0021,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Marker
      coordinate={
        Object {
          "latitude": 40.729713,
          "longitude": -73.9915285,
        }
      }
      image={1}
      style={
        Object {
          "position": "relative",
          "zIndex": 1,
        }
      }
    />
    <Marker
      coordinate={
        Object {
          "latitude": 40.71390377,
          "longitude": -74.00609437,
        }
      }
      id={1}
      onPress={[Function]}
    >
      <Callout
        onPress={[Function]}
        style={undefined}
        tooltip={true}
      />
    </Marker>
    <Marker
      coordinate={
        Object {
          "latitude": 40.89098508,
          "longitude": -73.89453102,
        }
      }
      id={2}
      onPress={[Function]}
    >
      <Callout
        onPress={[Function]}
        style={undefined}
        tooltip={true}
      />
    </Marker>
  </MapView>
</View>
`;
